src/main.h

```diff
@@ -191,8 +191,10 @@ static const uint64_t nMinDiskSpace = 52428800;
 /** Pruning-related variables and constants */
 /** True if any block files have ever been pruned. */
 extern bool fHavePruned;
-/** True if we're running in -prune mode. */
-extern bool fPruneMode;
+enum struct PruneMode {NONE=0, AUTO, MANUAL};
```

Comment:
Meta question: why do we need a tristate here? I think we could allow manual pruning even when in 'auto` mode. In that case, manual-only pruning could be requested by setting the limit very high.